### DISCUSSION TITLE
OpenAPI vendor extension

### DISCUSSION THREAD

**Discussion by nadir-albajari-hs** [apideck-libraries/portman#639] [Ideas]
Since some people autogenerate OpenAPI spec from source code, it will be nice to have vendor extensions for collection generation.

I've started implementing it on postman/openapi-to-collection and found your discontinued PR in postman's repository so also found this amazing repo

I've thought about a couple of extensions
- x-postman-collection-tests: [event1, event2]
- x-postman-collection-variables
- x-postman-environment-variable

**Comment by thim81** [comment]
hi @nadir-albajari-hs 

We thought about this direction, but we did go in that direction because:

The OpenAPI documents and describes the API.
The generated Postman collection serves to execute requests towards the API.

Adding OpenAPI vendor extensions would clutter the OpenAPI file and so also the documentation.
The OpenAPI vendor extension are also not really suited for handling variations or more complex configurations.

At the moment, the main focus is support the configuration option and not use OpenAPI vendor extension, since they serve different purposes.
